processing/src/main/java/org/apache/carbondata/processing/graphgenerator/GraphGenerator.java

```diff
@@ -998,4 +1000,24 @@ private void prepareIsUseInvertedIndex(List<CarbonDimension> dims,
     graphConfig.setIsUseInvertedIndex(
         isUseInvertedIndexList.toArray(new Boolean[isUseInvertedIndexList.size()]));
   }
+
+  /**
+   * Preparing the boolean [] to map whether the dimension use trim or not.
+   *
+   * @param dims
+   * @param graphConfig
+   */
+  private void prepareIsUseTrim(List<CarbonDimension> dims,
+                                GraphConfigurationInfo graphConfig) {
+    List<Boolean> isUseTrimList = new ArrayList<Boolean>();
+    for (CarbonDimension dimension : dims) {
+      if (dimension.isUseTrim()) {
```

Comment:
Can we add this trim option as property of the column, i mean inside a column property rather than setting directly as CarbonDimension property, Already in CarbonColumn there is a property map which holds column related properties, i think we can use that, please check the feasibility.